eva-accession-import-dbsnp2/src/main/java/uk/ac/ebi/eva/accession/dbsnp2/io/JsonNodeToClusteredVariantProcessor.java

```diff
@@ -73,6 +76,15 @@ private ClusteredVariant parseJsonNodeToClusteredVariant(JsonNode jsonRootNode)
                 continue;
             }
             String assemblyAccession = assemblyInfo.get(0).path("assembly_accession").asText();
+            // Ignore variant if it's assembly accession doesn't match the one supplied in input parameters
+            if(!StringUtils.equals(refseqAssembly, assemblyAccession)) {
```

Comment:
You don't need an auxiliary class for this, you could simply call `refseqAssembly.equals(assemblyAccession)`. The parameter must not be null, so that corner case is taking care of.

Reply:
corrected